pkg/agent/core/testhelpers/construct.go

```diff
@@ -36,12 +37,24 @@ type VmInfoOpt interface {
 	modifyVmInfoWithConfig(InitialVmInfoConfig, *api.VmInfo)
 }
 
+type ClockSourceOpt interface {
+	InitialStateOpt
+
+	clock() core.LogicClock
+}
```

Comment:
I think you don't need to add this if the clock is included in `core.Config` -- wdyt?

(no opinions, just noticed it seemed like a lot of effort here)